File: cpp/75.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low, mid, high;
        low = mid = 0; 
        high = nums.size()-1;
        while(mid <= high){
            if(nums[mid]==1){
                mid++;
            }else if(nums[mid]==0){
                swap(nums[mid], nums[low]);
                mid++;
                low++;
            }else{
                swap(nums[mid], nums[high]);
                high--;
            }
        }
    }
};

File: golang/75/75.go
package main

func sortColors(nums []int) {
	p0, p2 := 0, len(nums)-1
	for i := 0; i <= p2; i++ {
		for i <= p2 && nums[i] == 2 {
			nums[i], nums[p2] = nums[p2], nums[i]
			p2--
		}

		if nums[i] == 0 {
			nums[p0], nums[i] = nums[i], nums[p0]
			p0++
		}
	}

	return
}
